src/WebJobs.Script/Diagnostics/CategoryTraceWriterFactory.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal class CategoryTraceWriterFactory
+    {
+        private readonly string _category;
+        private readonly ScriptHostConfiguration _scriptHostConfig;
+
+        public CategoryTraceWriterFactory(string category, ScriptHostConfiguration scriptHostConfig)
```

Comment:
@mathewc previously expressed concern about these logs going into host logs. I think it would be more understandable to see these traces in host logs - you could infer function correlation if you see the console logs appear between a particular function started / function completed message.